### PR TITLE
Add zap logger support

### PR DESCRIPTION
**Pull request by NishanthSpShetty**
Add support for [uber/zap](https://github.com/uber-go/zap) logger to coldbrew logger.

### CODE REVIEW

File: loggers/zap/zap.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+type logger struct {
+	logger *zap.Logger
+	opt    loggers.Options
+	cfg    zap.Config
+}
+
+func (l *logger) Log(ctx context.Context, level loggers.Level, skip int, args ...interface{}) {
+
+	msg := fmt.Sprint(args...)
```
Comment (ankurs) [inline loggers/zap/zap.go]: should we add support for sugared and non sugared logger ?

Comment (ankurs) [inline loggers/zap/zap.go]: we should add ctxfields to the logs as well - https://github.com/go-coldbrew/log/blob/613591fa528737e04db94d1fc53e84832bba2b1d/loggers/gokit/gokit.go#L29

Comment (NishanthSpShetty) [reply loggers/zap/zap.go]: yes, we should let the application choose the type of zap logger they want, but the difference would be we won't be able to log data with the same reason of sugared logger, i.e to have logger without any overhead on identifying the types.

Comment (ankurs) [reply loggers/zap/zap.go]: we could add this as a loggers.Option right ?

File: loggers/zap/zap.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	if opt.JSONLogs {
+		zapCfg.Encoding = "json"
+	} else {
+		zapCfg.Encoding = "console"
+	}
+	l, err := zapCfg.Build()
+
+	if err != nil {
+		//should we fail? will use sugared log here
+		l, _ = zap.NewProduction()
```
Comment (ankurs) [inline loggers/zap/zap.go:116]: we should ideally return errors in NewLogger function as well, what do you think ?

Comment (NishanthSpShetty) [reply loggers/zap/zap.go:116]: yes, we have to change the interface, would be a lot of change for this one alone, so may be a separate PR to change all other logger implementation?

File: loggers/zap/zap.go (hunk 1 above)
Comment (NishanthSpShetty) [reply loggers/zap/zap.go]: yes, setting up this logger is easy, however, using it bit difficult. sugared logger is quite straightforward, non sugared logger expects typed fields, we pass `map[string]interface{}` to `log.Info`, which then we would have to convert by using type assertion which defeats the purpose of using the zap.

**Pull request merged**